lib/Mastodon/System/SessionCalls.vala

```diff
@@ -61,12 +61,12 @@ public partial class Backend.Mastodon.Session : Backend.Session {
    * This is an platform-specific implementation of the abstract method
    * defined in the base class, for more details see the base method.
    */
-  internal override Backend.Post load_post (Json.Object data) {
+  internal override Backend.Post load_post (Json.Object data, bool force_load = false) {
     // Get the id of the post
     string id = data.get_string_member ("id");
 
     // Check if the post is already present in memory
-    if (pulled_posts.contains (id)) {
+    if (pulled_posts.contains (id) && ! force_load) {
       return pulled_posts [id];
```

Comment:
Just a thought of mine:
I would like to update certain things like metrics later should they be pulled again.
So we need an update method later here.

Reply:
Looking at the structure of toots… I don't think it's possible to cache them _and_ update them. There's no "last modified" time that takes into account metrics. So our options appear to be:

1. use `created_at` (which never changes) and `edited_at` (which is only for an actual edit) and don't update metrics unless an edit happens
1. use timestamps and time-based caching so that we get some caching benefits but get inconsistent behaviour on whether metrics update after a like/boost depending on how long ago we cached the post
1. selectively bypass the cache with a "force load"
1. always load from JSON so that we get the latest metrics (but keep the cache so that we can update the existing object)

I'm going to go with "always load" and move to the updating and property binding.

Reply:
Yes, option 4 was what I had in mind.